### PR TITLE
Support different data serialisations

### PR DESCRIPTION
**Pull request by Vinnl**
We used to read and write one single serialisation for each data
type, but XML Schema actually defines multiple serialisations for
several types of data. For example, the boolean `true` can be
represented as both `"1"` and `"true"`. For more info, see
https://www.w3.org/TR/xmlschema-2/#built-in-datatypes

This commit introduces a couple of fixes to deal with the various
serialisations:

- The get*() functions on Things now understand the different
  serialisations. For example, this means that getBoolean() no
  longer returns `null` for the a boolean literal "false", but the
  expected value `false`.
- The remove*() functions on Things now remove all equivalent
  serialisations of a given value. For example, this means that
  passing `false` to removeBoolean() now removes both boolean
  literals with the value `"0"` (as before) as well as those with
  the value `"false"`.
- The serialisations we use by default have changed. Booleans are
  now `"true"` or `"false"` rather than `"1"` or `"0"`,
  respectively, and we now use JavaScript's native serialisations
  of Dates to ISO 8601 strings, rather than manually converting it
  and truncating milliseconds from it.

- [x] I've added a unit test to test for potential regressions of this bug.
- [x] The changelog has been updated, if applicable.
- [x] Commits in this PR are minimal and [have descriptive commit messages](https://chris.beams.io/posts/git-commit/).

### CODE REVIEW

**Comment by andydavison** [comment]
This fix seems to resolve reading true/false values.

I am however getting `409 Conflict` with editing a value, e.g. `The patch could not be applied. Could not find to delete: <https://andydavison.inrupt.net/profile/card#me> <http://xmlns.com/foaf/0.1/testboolean2> "false"^^<http://www.w3.org/2001/XMLSchema#boolean> .`

this is with a request payload
```
DELETE DATA {<https://andydavison.inrupt.net/profile/card#me> <http://xmlns.com/foaf/0.1/testboolean2> "false"^^<http://www.w3.org/2001/XMLSchema#boolean>.}; INSERT DATA {<https://andydavison.inrupt.net/profile/card#me> <http://xmlns.com/foaf/0.1/testboolean2> "true"^^<http://www.w3.org/2001/XMLSchema#boolean>.};
```

**Comment by Vinnl** [comment]
Thanks for testing @andydavison, I'll take a look at that.

**Comment by Vinnl** [comment]
OK, with help of @andydavison kindly sending me a recording reproducing the bug, @megoth verifying that the request we sent looks correct to him, and trying it out against a different server, I'm quite sure that the 409 Conflict is a bug in Node Solid Server: https://github.com/solid/node-solid-server/issues/1468

Thus, I'm marking this for ready for review again. I did add some additional tests that should verify that parsing/serialising works as expected (b0661c5e4600654928470e61e2c93e3f5588b362).

**Comment by Vinnl** [comment]
I also added 279e7026a43e387fee6a4f36b17454baadd494a7 that demonstrates the bug - it is currently disabled for NSS (where it fails). If it gets resolved there, or we want to test it against another server, we can simply use that test to verify it.

File: src/datatypes.test.ts [hunk 1]
```diff
@@ -43,13 +44,83 @@ import {
 } from "./datatypes";
 import { LocalNode } from "./interfaces";
 
+describe("stress-testing serialisations", () => {
+  it("should always return the input value when serialising, then deserialing a boolean", () => {
+    const runs = 100;
+    expect.assertions(runs + 2);
+
+    const fcResult = fc.check(
```
[the remark this answers is not in the document]
Comment (NSeydoux) [reply src/datatypes.test.ts:52]: Oh I wasn't aware of property-based testing, I'm currently reading https://dev.to/meeshkan/property-based-testing-for-javascript-developers-21b2, so maybe this comment is irrelevant

[the remark this answers is not in the document]
Comment (Vinnl) [reply src/datatypes.test.ts:52]: Hmm maybe stress-test was the wrong description; I was just getting afraid that I might have missed edge cases that would not serialise properly, with the additional bug reports. (And I did actually catch one: 1 Jan 1970 didn't get properly serialised, IIRC.)
